### DISCUSSION TITLE
Spiral Linux on Chromebooks/ Chromeboxes

### DISCUSSION THREAD

**Discussion by gabriele-3** [SpiralLinux/SpiralLinux-project#38] [Q&A] [answered]
Hi there.

This is gabriel-3 aka gabriel_3 aka gabriel3, 9+ years Linux user, former mod and member of openSUSE project.

While not being myself a Gecko Linux user I was recommending it to people new to Linux and to openSUSE because of its good quality.

In the recent past (2 years) have been running Linux on a Chromebook not any longer supported by Google, GalliumOS first, openSUSE later on (Leap 15.3 / Tumbleweed, Leap 15.4 tested only).

For a long time I was an openSUSE enthusiastic user, but Leap last release looks heavy for this job and Tumbleweed kills the poor machine when updating upon a full rebuild. In addition to that it's a bit hard to be approached by Linux newcomers.

Today I'm running and recommending PeppermintOS, since the last release Debian based as Spiral is, aimed to be minimal.

My question is: could be Spiral a good Chromebook operating system? 

The requirements are:

* Minimalism because of the reduced internal storage
* Lightness because of the low specs hardware
* Easiness of install/use because many people looking for Linux on unsupported Chromebooks are new to Linux.

As an example, while I love `snapper` and its rollback feature, I'm quite worried about the storage needed for the snapshots, on the other (bright) hand I run  PeppermintOS on a compressed btrfs file system (`compress-force=zstd`), similar to the Spiral set up.

**Comment by geckolinux** [comment]
Hey Gabriel, great to see you here!

It sounds like we're in a similar situation, I also need my OS to be viable for my personal needs as well as being approachable and easy enough for new Linux users that I support. I have found GeckoLinux and SpiralLinux to be best for that, and I think that vanilla openSUSE and Debian would be very poor choices for most new users that aren't familiar with Linux.

So, in terms of lightness, I would say that GeckoLinux and SpiralLinux are extremely similar in terms of RAM usage. They both boot up and login to the desktop with a significantly lower memory footprint than many other distros using the same desktop environment. This is simply because I make a point of not pre-installing any welcome windows or automatic package updaters.

For old hardware I've personally found that Mate tends to be the best balance between good performance and usability. It actually feels lighter than XFCE in my subjective experience. LXQt is lighter still, but it has some quirks that I personally don't like.

I would say that SpiralLinux in several aspects is actually a bit more user friendly than GeckoLinux. But since you're a long-time openSUSE like me you may still prefer GeckoLinux for your own systems and/or for new users that you support. But do take into account the fact that openSUSE Leap as we know it will cease to exist in a few years according to openSUSE's plans. Until then GeckoLinux Static will continue to be an option. I very much agree that openSUSE Tumbleweed and GeckoLinux Rolling are usually not a good idea for new users due to the constant change and potential for breakage. But I did have a case last year where a new user got a laptop with a new audio chipset that wasn't supported by the Leap kernel, so I had to recommend they install GeckoLinux Rolling, and it actually went fine. But now with SpiralLinux I think it offers the best of both worlds-- a slow-changing base system with the option of using the much newer kernel and firmware from Debian backports.

Snapper doesn't necessarily have to take a lot of space, especially if the OS doesn't get updated frequently as is the case for GeckoLinux Static and SpiralLinux. You could also reduce the number of snapshots, and Snapper will automatically clean them up. To do that in openSUSE you need to adjust it manually in `/etc/snapper/configs/`. In SpiralLinux you can use **Snapper GUI** to adjust the config. But if you don't want to deal with Btrfs for yourself or for other users you support, the Calamares configuration in both GeckoLinux and SpiralLinux also makes it super easy to use EXT4 or XFS.

I hope you can find a good solution for your needs and other users. Let me know how SpiralLinux works for you!

**Reply by NikkiSatmaka** [reply]
Hi Sam,

I've just installed SpiralLinux using EXT4. Is it safe to uninstall snapper? Or what other packages are not needed since I'm not using BTRFS?

**Reply by geckolinux** [reply]
Hi there, yep, you can go ahead and uninstall Snapper and the Snapper GUI packages. You could also remove `btrfs-progs` and `python3-btrfsutil` but they're really quite small and I wouldn't bother with them. You could delete the `/etc/grub.d/41_snapshots-btrfs` file too, which is what is normally responsible for generating the GRUB menu for the Btrfs snapshots, but it also shouldn't cause any problems if you leave it.

**Reply by NikkiSatmaka** [reply]
If I remove those packages, will it be automatically reinstalled at some point because of some updates? Like if there's a hook that checks them?

**Reply by geckolinux** [reply]
> If I remove those packages, will it be automatically reinstalled at some point because of some updates? Like if there's a hook that checks them?

Nope.

**Comment by gabriele-3** [comment]
Hi Sam and thank you for answering.

Geckois out of this game, either for myself or for Chromebooks refugees
unless they are explicitly willing to go with openSUSE.

Spiral:  I'll give a try to the Xfce version in order to compare the same
desktop environment with the current system.

Cheers,

gabriel_3

**Reply by geckolinux** [reply]
> Spiral: I'll give a try to the Xfce version in order to compare the same desktop environment with the current system.

Please do let me know how it went when you find time. Thanks!

**Comment by gabriele-3** [comment, marked as answer]
I did a quick test.

Overall it's a good distro, with sane and advanced settings:
* Zram
* Compressed btrfs
* Backported kernel available for newer hardware

However, related to the specific task of ChromeOS substitution on
Chromebooks done by newcomers:

* it's not minimal e.g. Thunderbird and LibreOffice are preinstalled
* it is visually unpleasant
* snapper and its snapshots are life savers for tinkerer, but almost
unuseful on Debian
* sound is not working out of the box with my specific hardware, Debian does

Am I going to switch from PeppermintOS? No for the time being.

Will I keep an eye on this project to see how it grows? Yes, it looks
promising.

Overall I think it's a perfect backup plan for Gecko Linux if openSUSE will
drop the desktop use case.

gabriel_3

Il Lun 4 Lug 2022, 15:03 GeckoLinux ***@***.***> ha scritto:

**Reply by geckolinux** [reply]
> sound is not working out of the box with my specific hardware, Debian does

This would be the most serious concern. I'd appreciate some more details on your hardware, the specific ISO version that you tested, and a screenshot of the `pavucontrol` **Configuration** tab.

**Reply by gabriele-3** [reply]
Unfortunately I cannot assit you with the debug.

I just restored the disk image I backed up before testing Spiral and I do not have Spiral any longer on my system.

Some pieces of info:

- sound card chtmax98090, `cat /proc/asound/card*` output showed it
- volume control showed *dummy output*

Even if it looks unrelated to me, LMDE has the same issue and it's the only other Debian based distro I tested with this issue.

**Reply by geckolinux** [reply]
Thanks for the information. You could just test it with the live USB. Which SpiralLinux ISO did you try?

**Reply by geckolinux** [reply]
The soundcard model was very helpful, and I found one of your previous threads:
https://www.reddit.com/r/GalliumOS/comments/ixjrs8/gallimuos_next_release_baytrail_with_sound_card
Looks like the `alsa-ucm-conf` package needs to be included for your hardware. I'll look into pre-installing it for the next release.

**Reply by gabriele-3** [reply]
iso: SpiralLinux_XFCE_11.220628_x86-64.iso

I confirm the fix: I installed `alsa-ucm-conf `on the live system, logged out and in again and sound works as expected.

**Reply by geckolinux** [reply]
Awesome, thanks very much for the help with this, that will be an important improvement for SpiralLinux.

**Reply by geckolinux** [reply]
Implemented  in SpiralLinux 11.220925 release.
